chain-signatures/node/src/mesh/connection.rs

```diff
@@ -83,7 +101,12 @@ impl NodeConnection {
                 Ok(state) => {
                     let new_status = match state {
                         StateView::Running { .. } | StateView::Resharing { .. } => {
-                            NodeStatus::Active
+                            // The peer is running. But before we can reliably
+                            // use the connected node in protocols we initiate,
+                            // we need to ensure the peer has the up-to-date
+                            // data about out owned IDs.
+                            NodeStatus::Syncing
```

Comment:
hmm, wouldn't this always trigger a sync on a ping?

Reply:
True! I think I just missed a check here... Let me fix that quickly

Reply:
Should be fixed now